x-pack/plugins/security_solution/server/lib/detection_engine/rules/read_rules.ts

```diff
@@ -38,20 +37,26 @@ export const readRules = async ({
         throw err;
       }
     }
-  } else if (ruleId != null) {
+  } else if (ruleIds != null && ruleIds.length > 0) {
     const ruleFromFind = await findRules({
       alertsClient,
-      filter: `alert.attributes.tags: "${INTERNAL_RULE_ID_KEY}:${ruleId}"`,
+      filter: ruleIds
+        .map((ruleId) => `alert.attributes.tags: "${INTERNAL_RULE_ID_KEY}:${ruleId}"`)
+        .join(' OR '),
       page: 1,
       fields: undefined,
       perPage: undefined,
       sortField: undefined,
       sortOrder: undefined,
     });
-    if (ruleFromFind.data.length === 0 || !isAlertType(ruleFromFind.data[0])) {
-      return null;
+    const rules = ruleFromFind.data;
+    if (
+      ruleFromFind.data.length > 0 &&
+      rules.every((rule): rule is RuleAlertType => isAlertType(rule))
+    ) {
+      return rules;
```

Comment:
I also debated returning an array of found rules or an object of found rules where the keys are the ruleIds. Open to hearing opinions on this too.

Reply:
I was just thinking since we do iterate over the array returned by the `readRules` function pretty heavily it might make things faster if each `.find` was replaced with `foundRuleObject[myruleId]` which would result in faster checking for the existence of the rule returned by this function. I think that would be cleaner.